strawberry/aiohttp/views.py

```diff
@@ -166,9 +166,9 @@ async def handle_stop(self, message: dict) -> None:
         await self.cleanup_operation(operation_id)
 
     async def handle_keep_alive(self, ws: web.WebSocketResponse) -> None:
-        await asyncio.sleep(self.keep_alive_interval)
-        await ws.send_json({"type": GQL_CONNECTION_KEEP_ALIVE})
-        await self.handle_keep_alive(ws)
+        while True:
+            await asyncio.sleep(self.keep_alive_interval)
+            await ws.send_json({"type": GQL_CONNECTION_KEEP_ALIVE})
```

Comment:
oh, so the recursion was happening when the code was left running for a long time, right?

Reply:
should we do the same here? https://github.com/strawberry-graphql/strawberry/blob/aa3ab0e51a25a7b2b0c664820187c82d2783e27b/strawberry/asgi/__init__.py#L70-L79

Reply:
> oh, so the recursion was happening when the code was left running for a long time, right?

exactly

> should we do the same here? 

I don't think the ASGI implementation has the same problem since it's using `asycio.create_task` which is non blocking as far as I know.

(However, I planned to refactor the ASGI subscription implementation in a separate PR to bring both implementations closer together and increase ASGI test coverage.)

Reply:
gotcha, thanks for explaining!